invoices/syncing/tenkfeet.py

```diff
@@ -210,119 +238,189 @@ def match_user(self, email):
+                    logger.debug("Skipping hour entry with 0 incurred hours: %s", entry)
+                else:
+                    entry_date = entry["date"]
+                    entry['fo0'] = "bar"
+                    date_data[entry_date]["sha256"].update(json.dumps(entry, cls=DateTimeEncoder).encode())
+                    date_data[entry_date]["items"].append(entry)
+
+            return date_data
+
+        def merge_data(entry, users, assignables):
+            def datetime_to_date(dt):
+                if dt:
+                    return dt.date()
+                return None
+
+            def legacy_leave_type():
+                if entry["assignable_type"] == "Project":
+                    return "[project]"
+                return entry["assignable_type"]
+
+            def status(approval):
+                if not approval:
+                    return "Unsubmitted"
+                if approval.get("approved_at"):
+                    return "Approved"
+                return "Pending Approval"
+
+            approvals = entry.get("approvals") or {}
+            approval_data = approvals.get("data")
+            if approval_data:
+                approval = approval_data[0]
+            else:
+                approval = {}
+
+            user = users[entry["user_id"]]
+            assignable = assignables.get(entry["assignable_id"]) or {}
+            project = assignable.get("project") or {}
+
+            data = {
+                "date": entry["date"],
+                "user_id": entry["user_id"],
+                "user_name": user["display_name"],
+                "assignable_id": entry["assignable_id"],
+                "approved_at": datetime_to_date(approval.get("approved_at")),
+                "upstream_id": entry["id"],
+                "approved_by": approval.get("approved_by"),
+                "submitted_by": approval.get("submitted_by"),
+                "updated_at": entry.get("updated_at"),
+                "created_at": entry.get("created_at"),
+                "project": project.get("name") or "[Leave Type]",
+                "client": project.get("client") or "[none]",
+                "incurred_hours": entry["hours"],
+                "incurred_money": entry["bill_rate"] * entry["hours"],
+                "category": entry.get("task") or "[none]",
+                "notes": entry.get("notes"),
+                "entry_type": "Confirmed" if not entry["is_suggestion"] else "Suggestion",
+                "discipline": user.get("discipline") or "[none]",
+                "role": user.get("role") or "[none]",
+                "bill_rate": entry["bill_rate"],
+                "leave_type": legacy_leave_type(),
+                "phase_name": assignable.get("phase", {}).get("phase_name") or "[Non Phase Specific]",
+                "billable": is_phase_billable(
+                    phase_name="",
+                    project=project.get("name") or "[Leave Type]"
+                ),  # TODO
```

Comment:
TODO what?

Reply:
check what the hell is `billable` and is that info even relevant in any way, most likely that is parameter on user because that is only place where it is mentioned in API docs.